### PR TITLE
feat(docs): update how-to-writer-xxx-functions docs

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

## Changelog

- Improvement
- Documentation
 

## Related Issues

Fixes #issue

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Review by BohuTANG — APPROVED** [review]
looks good here

File: docs/doc/10-faq/06-how-to-write-scalar-functions.md [hunk 1]
```diff
@@ -128,66 +60,167 @@ Would look like this:
...
+    ```rust
+    #[derive(Clone)]
+    pub struct ConstColumn {
+        length: usize,
+        column: ColumnRef,
+    }
+    ```
+- Nullable column
+
+    By defaults, column are not nullable. If we want nullable column, we can use this to represent it.
```
Comment (Veeupup) [inline docs/doc/10-faq/06-how-to-write-scalar-functions.md:83]: typo to:
By default, columns are not nullable. If we want a nullable column, we can use this to represent it.

File: docs/doc/10-faq/06-how-to-write-scalar-functions.md [hunk 2]
```diff
@@ -128,66 +60,167 @@ Would look like this:
...
+So inside the `eval` function, we really don't need to care about constant or nullable cases. It's pretty simple and efficient.
+
+
+The macro `with_match_primitive_type_id` will match the primitive type id, and cast the column into corresponding type, so we allowed `sqrt(i8)`, `sqrt(i16)` ... types.
+
+The `scalar_unary_op` is a helper function to implement the scalar function for unary operator.
+This is very common used and there is `scalar_binary_op` too. See more in [binary](https://github.com/datafuselabs/databend/blob/e7edeea2e3ae5fb1f8408903df10b1b641b57652/common/functions/src/scalars/expressions/binary.rs), [unary](https://github.com/datafuselabs/databend/blob/e7edeea2e3ae5fb1f8408903df10b1b641b57652/common/functions/src/scalars/expressions/unary.rs)
```
Comment (Veeupup) [inline docs/doc/10-faq/06-how-to-write-scalar-functions.md:187]: common -> commonly

**Pull request merged**